### PR TITLE
Fuse code

### PR DESCRIPTION
**Pull request by whalelephant**
- moves offchain-api-clients code here
- update `src/offchain-ipfs-rust.md` to include code 
- _attempt_ update .github/work_flows to include running the code without exit codes

### CODE REVIEW

**Review by aphelionz — APPROVED** [review]
LGTM! I'll let @ljedrz comment on the Rust code itself if he wants, otherwise you are free to rebase + squash + merge

**Comment by aphelionz** [comment]
Also: I added linting to the `main` branch so that's why there are conflicts. Sorry about that!

To run linting locally you can run `npx markdownlint-cli src/ READMEmd`

File: code/sub-api-ipfs/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    let genesis_hash = api.genesis_hash;
+    let spec_version = api.runtime_version.spec_version;
+    let transaction_version = api.runtime_version.transaction_version;
+
+    // Example bytes to add
+    let bytes_to_add: Vec<u8> = vec![1, 2, 3, 4];
+    // Example CID for the example bytes added vec![1, 2, 3, 4]
+    let cid = String::from("QmRgctVSR8hvGRDLv7c5H7BCji7m1VXRfEE1vW78CFagD7")
+        .into_bytes()
+        .to_vec();
```
Comment (ljedrz) [inline code/sub-api-ipfs/src/main.rs]: this line should be redundant after `into_bytes()`

File: code/sub-api-ipfs/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    // Example multiaddr to connect IPFS with
+    let multiaddr = String::from(
+        "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+    )
+    .into_bytes()
+    .to_vec();
```
Comment (ljedrz) [inline code/sub-api-ipfs/src/main.rs]: ditto; hasn't `clippy` complained here?

File: code/sub-api-ipfs/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    // Example Peer Id
+    let peer_id = String::from("QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ")
+        .into_bytes()
+        .to_vec();
```
Comment (ljedrz) [inline code/sub-api-ipfs/src/main.rs]: same

File: code/subxt-ipfs/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    // Example multiaddr to connect IPFS with
+    let multiaddr = String::from(
+        "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+    )
+    .into_bytes()
+    .to_vec();
+    // Example Peer Id
+    let peer_id = String::from("QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ")
+        .into_bytes()
+        .to_vec();
```
Comment (ljedrz) [inline code/subxt-ipfs/src/main.rs]: same 3 redundant `to_vec()` conversions in the 3 assignments above

File: code/sub-api-ipfs/src/main.rs (hunk 2 above)
Comment (whalelephant) [reply code/sub-api-ipfs/src/main.rs]: strange.. no it did not for me.

File: code/sub-api-ipfs/src/main.rs (hunk 1 above)
Comment (whalelephant) [reply code/sub-api-ipfs/src/main.rs]: 👌

**Pull request merged**